### PR TITLE
Write Some Assertions for Validation Library

### PR DESCRIPTION
**Pull request by sohelrana820**

### CODE REVIEW

File: tests/TestCase/Validation/ValidationTest.php [hunk 1]
```diff
@@ -2813,6 +2817,9 @@ public function testContainNonAlphaNumeric()
         $this->assertFalse(Validation::containsNonAlphaNumeric('abcdef##', 3));
         $this->assertFalse(Validation::containsNonAlphaNumeric('abc##def', 3));
         $this->assertFalse(Validation::containsNonAlphaNumeric('ab#cd#ef', 3));
+
+        //CHECK SCALAR
```
Comment (markstory) [inline tests/TestCase/Validation/ValidationTest.php]: Why are these comments in ALL CAPS?

Comment (sohelrana820) [inline tests/TestCase/Validation/ValidationTest.php]: No special reason. Should I make it capitalize or lowercase?

Comment (dereuromark) [inline tests/TestCase/Validation/ValidationTest.php]: Do we need them at all?

Comment (sohelrana820) [inline tests/TestCase/Validation/ValidationTest.php]: I think so. Cause someone might be tried with Scalar value (CC or AlphaNumeric).

Comment (dereuromark) [inline tests/TestCase/Validation/ValidationTest.php]: But arent you doing the opposite here? Checking a non-scalar (array) value?
Usually one can make a separate test method with the method name making it clear, also optionally adding a doc block note there.

Comment (sohelrana820) [inline tests/TestCase/Validation/ValidationTest.php]: So, what do you suggest here?

Comment (dereuromark) [inline tests/TestCase/Validation/ValidationTest.php]: testContainNonAlphaNumericWithArray() etc for example.

Comment (sohelrana820) [inline tests/TestCase/Validation/ValidationTest.php]: OK... Please do me a favor. I will do something like that you said. So should I send another pull request after modify it? And If so then what about my current pull request?

Comment (dereuromark) [inline tests/TestCase/Validation/ValidationTest.php]: You can add commits to your branch. This PR will automatically be updated :)

File: tests/TestCase/Validation/ValidationTest.php [hunk 2]
```diff
@@ -3104,4 +3099,10 @@ public function testImageWidth()
         $this->assertFalse(Validation::imageWidth($upload, '>', 2000));
         $this->assertFalse(Validation::imageWidth($upload, '==', 3000));
     }
+
+    public function testContainNonAlphaNumericAndCCWithScalar()
```
Comment (dereuromark) [inline tests/TestCase/Validation/ValidationTest.php]: Why scalar? Those are non scalar values - also the 2nd one is not the same as in the previous commit (it now is a scalar).

File: tests/TestCase/Validation/ValidationTest.php [hunk 3]
```diff
@@ -168,6 +168,7 @@ public function testLengthBetween()
 
         $this->assertFalse(Validation::lengthBetween('abcdefg', 1, 6));
         $this->assertFalse(Validation::lengthBetween('ÆΔΩЖÇ', 1, 3));
+        $this->assertFalse(Validation::lengthBetween(1, 1, 3));
```
Comment (dereuromark) [inline tests/TestCase/Validation/ValidationTest.php]: I feel like this should be true actually - is there maybe a cast missing? Or are we ok with the string input only for it?

File: tests/TestCase/Validation/ValidationTest.php [hunk 4]
```diff
@@ -3097,4 +3099,10 @@ public function testImageWidth()
         $this->assertFalse(Validation::imageWidth($upload, '>', 2000));
         $this->assertFalse(Validation::imageWidth($upload, '==', 3000));
     }
+
+    public function testContainNonAlphaNumericAndCCWithScalar()
+    {
+        $this->assertFalse(Validation::cc(['869972521242198'], ['voyager']));
+        $this->assertFalse(Validation::containsNonAlphaNumeric('ab#cd#ef', 3));
```
Comment (markstory) [inline tests/TestCase/Validation/ValidationTest.php]: Why is this here vs. with the other tests for this method?

File: tests/TestCase/Validation/ValidationTest.php (hunk 2 above)
Comment (sohelrana820) [inline tests/TestCase/Validation/ValidationTest.php]: I will fix it.

File: tests/TestCase/Validation/ValidationTest.php (hunk 3 above)
Comment (sohelrana820) [inline tests/TestCase/Validation/ValidationTest.php]: No casting issue. It's just take string input.

File: tests/TestCase/Validation/ValidationTest.php (hunk 4 above)
Comment (sohelrana820) [inline tests/TestCase/Validation/ValidationTest.php]: So...What do you suggest?

Comment (markstory) [inline tests/TestCase/Validation/ValidationTest.php]: These tests could be added to the existing test methods.

Comment (sohelrana820) [inline tests/TestCase/Validation/ValidationTest.php]: OK... I will modify it.

**Comment by dereuromark** [comment]
I am still not sure any of the tests brings value. They are either covered or completely out of scope (passing an array never works by definition).

**Comment by markstory** [comment]
Thanks @sohelrana820
